### ISSUE TITLE
Feature request: Ruby runtime

### ISSUE THREAD

**Issue by j4zzcat** [apache/incubator-openwhisk#2046]
Make Ruby a first-class citizen in OpenWhisk.

**Comment by cclauss** [comment]
A few key steps to get this ball rolling:
- [ ] a [`hello.rb` script](https://github.com/openwhisk/openwhisk/tree/master/tests/dat/actions) that takes in a dict (hash ?) and returns a modified dict
- [ ] a [Dockerfile](https://github.com/openwhisk/openwhisk/tree/master/core) that builds a reasonable Ruby version
- [ ] a related [build.gradle](https://github.com/openwhisk/openwhisk/blob/master/core/pythonAction/build.gradle) to drive the docker build

If we could get those three things done, we could evaluate the feasibility of this request.

**Comment by j4zzcat** [comment]
hello.rb
```
def main( input )                                                                                                               
  if name = input[ 'name' ]                                                                                                     
    { 'greeting' => "Hello #{name}!" }                                                                                          
  else                                                                                                                          
    { 'greeting' => 'Hello stranger!' }                                                                                         
  end                                                                                                                           
end      
```

Dockerfile
```
FROM ruby:2.4.1                                                                                                                 
RUN  gem install \                                                                                                              
       bundler \                                                                                                                
       sinatra \                                                                                                                
       activesupport \                                                                                                          
       yaml-ostruct \                                                                                                           
       couchrest \                                                                                                              
       jwt        
```

build.gradle
```
ext.dockerImageName = 'rubyAction'                                                                                              
apply from: '../../gradle/docker.gradle'                                                                                        
distDocker.dependsOn ':core:actionProxy:distDocker'
```

**Comment by csantanapr** [comment]
Fixed by #3725
